### PR TITLE
Build pip wheel based on conan installation

### LINKED ISSUE #292: Lanelet2 pip wheel incompatible with `GLIBC_2.28`
The pypi wheel currently requires `GLIBC_2.29` and `auditwheel` is not able to "repair" it to `2_28` if changed here:
https://github.com/fzi-forschungszentrum-informatik/Lanelet2/blob/11e83fdd208e06fe1131cbf5aeaea7586a3d0a99/.github/workflows/cd.yaml#L73

Workable solution is a conan build as in https://github.com/jdrudolph/Lanelet2/blob/bb64b1a118c428c759cf86d7a48d58a5466fe4cc/Dockerfile_conan_cp38

### PR DESCRIPTION
**Pull request by m-naumann**
closes #292 

starting point for #287

### CODE REVIEW

**Comment by jdrudolph** [comment]
> When I try to install the resulting wheel in an Ubuntu 18.04 docker, I still get the error
> 
> `ERROR: lanelet2-1.2.0-cp38-cp38-manylinux_2_28_x86_64.whl is not a supported wheel on this platform.`
> 
> `auditwheel show` however claims the wheel is indeed `manylinux_2_28` compatible and should therefore work with 18.04. Did you test that yet?
> 
> (Note: `pip` is already upgraded to version 23)

According to https://packages.ubuntu.com/source/bionic/glibc the `glibc` version in 18.04 is `2.27`. You should also be able to verify this by running e.g. `ldd --version`. So I would consider this to be expected.

In theory I don't see any issues with potentially using an even older toolchain in the `conanio/gccX` container. It's just a matter of trying these things out.

I'm currently testing a matrix build for Python version 3.8, 3.9, 3.10, 3.11. However compilation for >3.10 is failing.
* 3.11 some weird boost python error https://github.com/jdrudolph/Lanelet2/actions/runs/4813135779/jobs/8569269317#step:4:1299
* 3.10 some even weirder config error https://github.com/jdrudolph/Lanelet2/actions/runs/4813135779/jobs/8569269170#step:4:1754

maybe there is an easy fix for those?

**Comment by jdrudolph** [comment]
it's working for Python 3.8 and 3.9 on both Ubuntu 20.04 and 22.04.
https://github.com/jdrudolph/Lanelet2/actions/runs/4813560152

it would be great to have support for 3.10 and 3.11 as well. in my opinion this is ready to merge as it's unclear to me what effort will be involved in making it work.

File: .github/conan_dockerfile/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  && echo "Lanelet2 version: $LANELET2_VERSION" \
+  && conan install lanelet2/$LANELET2_VERSION@lanelet2/stable --build=missing -g virtualenv
+
+FROM lanelet2_conan as lanelet2_conan_with_pip_wheel
+
+SHELL ["/bin/bash", "-c"]
+WORKDIR /home/conan
+RUN source /home/conan/src/lanelet2/activate.sh \
+  && LETAS_PACKAGE_DIR=$(python -c "import lanelet2; from pathlib import Path; print(Path(lanelet2.__file__).parent)") \
+  && cp -r $LETAS_PACKAGE_DIR .
```
Comment (m-naumann) [inline .github/conan_dockerfile/Dockerfile:52]:
```suggestion
  && LANELET2_PACKAGE_DIR=$(python -c "import lanelet2; from pathlib import Path; print(Path(lanelet2.__file__).parent)") \
  && cp -r $LANELET2_PACKAGE_DIR .
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -176,3 +176,20 @@ jobs:
...
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build dependencies
+        uses: docker/build-push-action@v3
+        with:
+          builder: ${{ steps.buildx.outputs.name }}
+          file: .github/conan_dockerfile/Dockerfile
+          tags: lanelet2_conan_with_pip_wheel
+          load: true
+          target: lanelet2_conan_with_pip_wheel
```
Comment (m-naumann) [inline .github/workflows/ci.yaml:195]:
```suggestion
          target: lanelet2_conan_with_pip_wheel
          build-args: PYTHON_VERSION=${{ matrix.python-version }}
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -176,3 +176,20 @@ jobs:
           coverage-files: ./lcov/full_coverage.lcov
           artifact-name: code-coverage-report
           github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  full_conan_installation:
+    needs: lint
```
Comment (m-naumann) [inline .github/workflows/ci.yaml:181]:
```suggestion
    needs: lint
    strategy:
       matrix:
         # builds for 3.10 and 3.11 are not yet working
         # python-version: ["3.8", "3.9", "3.10", "3.11"]
         python-version: ["3.8", "3.9"]
```

**Review by m-naumann — COMMENTED** [review]
🚀 Let's also test the matrix in CI

**Comment by immel-f** [comment]
I investigated support for 3.10 but could not get it running on my system. The error is related to `find_package(Python3 REQUIRED)`in the `mrt_cmake_modules`, which for some reason works with python 3.9 but not 3.10. It is probably related to CMake/`mrt_cmake_modules` (a CMake helper package used in Lanelet2) not working well with pyenv and should be checked out separately. 

Using `conanio/gcc7:latest`, the build works and the manylinux requirement can be dropped to 2_27, meaning it should then also work on Ubuntu 18.04 :+1:

One think of note is that GCC 7 in conanio uses Ubuntu 17 as its base in the `gcc7:latest` version. 
There also exist updated versions of the conanio dockers for Ubuntu 18 with gcc 7, but I think those do not setup pyenv properly as they throw a lot of linking errors during the build with any python version (meaning also 3.8 and 3.9).

The saving grace is that at least gcc 7 works, but after adapting that I would just merge this now as you suggested and leave python 3.10/3.11 for the future

**Comment by immel-f** [comment]
The 3.11 error btw is boost_python not compiling anymore in conan, which is probably related to the CPython changes in 3.11. I think this could be fixed by using a newer boost version, but lanelet2 is currently not compatible with boost 1.75+ (see https://github.com/fzi-forschungszentrum-informatik/Lanelet2/issues/211)

**Review by immel-f — APPROVED** [review]

**Pull request merged**